src/main/java/no/sikt/nva/pubchannels/channelregistry/ChannelRegistryClient.java

```diff
@@ -106,36 +106,37 @@ private <T> T executeRequest(HttpRequest request, Class<T> clazz)
         var response = httpClient.send(request, BodyHandlers.ofString());
 
         if (!OK_STATUSES.contains(response.statusCode())) {
-            handleError(response);
+            handleError(request.uri(), response);
         }
 
         return attempt(() -> dtoObjectMapper.readValue(response.body(), clazz)).orElseThrow();
     }
 
-    private void handleError(HttpResponse<String> response) throws ApiGatewayException {
+    private void handleError(URI requestedUri, HttpResponse<String> response) throws ApiGatewayException {
         var statusCode = response.statusCode();
         if (HTTP_NOT_FOUND == statusCode) {
+            LOGGER.error("Publication channel not found: {} {}", requestedUri, response.body());
             throw new NotFoundException("Publication channel not found!");
         }
         if (HTTP_BAD_REQUEST == statusCode) {
             throw new BadRequestException(response.body());
         }
         if (HTTP_MOVED_PERM == statusCode) {
             var location = response.headers().map().get("Location").getFirst();
-            LOGGER.info("Publication channel moved permanently to: {}", location);
+            LOGGER.info("Publication channel {} moved permanently to: {}", requestedUri, location);
             throw new PublicationChannelMovedException("Publication channel moved permanently!", URI.create(location));
         }
-        LOGGER.error("Error fetching publication channel: {} {}", statusCode, response.body());
+        LOGGER.error("Error fetching publication channel: {} {} {}", requestedUri, statusCode, response.body());
         throw new BadGatewayException("Unexpected response from upstream!");
     }
 
     private ApiGatewayException logAndCreateBadGatewayException(URI uri, Exception e) {
-        LOGGER.error("Unable to reach upstream: {}", uri, e);
         if (e instanceof InterruptedException) {
             Thread.currentThread().interrupt();
         } else if (e instanceof ApiGatewayException) {
             return (ApiGatewayException) e;
         }
+        LOGGER.error("Unable to reach upstream: {}", uri, e);
         return new BadGatewayException("Unable to reach upstream!");
```

Comment:
Is this reachable? `BadGatewayException` appears to extend `ApiGatewayException`, so it should never get past line 137?


```
var foo = new BadGatewayException("Unable to reach upstream!");
var bar = (foo instanceof ApiGatewayException);
// Linter warning:
// "Condition 'foo instanceof ApiGatewayException' is always 'true' "
```

Also, should we log `InterruptedException` or just ignore it?

Reply:
From what I understand here, anything that is not already an ApiGatewayException (including the InterruptedException) will be logged and expressed as a BadGatewayException. Not sure if I think this is a brilliant idea as there are probably things that aren't Bad-Gateway related that get sluiced into this catch-all.

Reply:
@brinxmat is right, anything that is not already an ApiGatewayException will be logged and expressed as a BadGatewayException. I.e. `ConnectException`

I can change this to treat it as an unexpected error if you think this is more correct?
```
private ApiGatewayException logUnexpectedError(URI uri, Exception e) {
      if (e instanceof InterruptedException) {
          LOGGER.error("Thread interrupted when fetching: {}", uri, e);
          Thread.currentThread().interrupt();
      } else if (e instanceof ApiGatewayException) {
          return (ApiGatewayException) e;
      }
      LOGGER.error("Unexpected error occurred when fetching: {}", uri, e);
      throw new RuntimeException(e);
  }
```

Reply:
This PR does not change todays implementation, just trying to avoid misleading logging

Reply:
Ah, I see! Ignore my comment then and do what you think is best :)